doc/ansible/dev/developer_guide.md

```diff
@@ -0,0 +1,430 @@
+`deploy/cr.yaml`:
+```yaml
+apiVersion: "foo.example.com/v1alpha1"
+kind: "Foo"
+metadata:
+  name: "example"
+```
+
+Since `spec` is not set, Ansible is invoked with no extra variables. The next
+section covers how extra variables are passed from a Custom Resource to
+Ansible. This is why it is important to set sane defaults for the operator.
+
+Create a Custom Resource instance of Foo with default var `state` set to
+`present`:
+```bash
+$ kubectl create -f deploy/cr.yaml
+```
+
+Check that namespace `test` was created:
+```bash
+$ kubectl get namespace
+NAME          STATUS    AGE
+default       Active    28d
+kube-public   Active    28d
+kube-system   Active    28d
+test          Active    3s
+```
+
+Modify `deploy/cr.yaml` to set `state` to `absent`:
+```yaml
+apiVersion: "foo.example.com/v1alpha1"
+kind: "Foo"
+metadata:
+  name: "example"
+spec:
+  state: "absent"
+```
+
+Apply the changes to Kubernetes and confirm that the namespace is deleted:
+```bash
+$ kubectl apply -f deploy/cr.yaml
+$ kubectl get namespace
+NAME          STATUS    AGE
+default       Active    28d
+kube-public   Active    28d
+kube-system   Active    28d
+```
+
+### Testing an Ansible operator on a cluster
+
+Now that a developer is confident in the operator logic, testing the operator
+inside of a pod on a Kubernetes cluster is desired. Running as a pod inside a
+Kubernetes cluster is preferred for production use.
+
+To build the `foo-operator` image and push it to a registry:
+```
+$ operator-sdk build quay.io/example/foo-operator:v0.0.1
+$ docker push quay.io/example/foo-operator:v0.0.1
+```
+
+Kubernetes deployment manifests are generated in `deploy/operator.yaml`. The
+deployment image in this file needs to be modified from the placeholder
+`REPLACE_IMAGE` to the previous built image. To do this run:
+```
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/foo-operator:v0.0.1|g' deploy/operator.yaml
+```
+
+Deploy the foo-operator:
+
+```sh
+$ kubectl create -f deploy/rbac.yaml
+$ kubectl create -f deploy/operator.yaml
+```
+
+**NOTE**: `deploy/rbac.yaml` creates a `ClusterRoleBinding` and assumes we are
+working in namespace `default`. If you are working in a different namespace you
+must modify this file before creating it.
+
+Verify that the foo-operator is up and running:
+
+```sh
+$ kubectl get deployment
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+foo-operator       1         1         1            1           1m
+```
+
+## Extra vars sent to Ansible
+The extravars that are sent to Ansible are predefined and managed by the operator. The `spec` section will pass along the key-value pairs as extra vars. This is equivalent to how above extra vars are passed in to `ansible-playbook`.
+
+For the CR example:
+```yaml
+apiVersion: "app.example.com/v1alpha1"
+kind: "Database"
+metadata:
+  name: "example"
+spec:
+  message:"Hello world 2"
+  newParameter: "newParam"
+```
+
+The structure is:
```

Comment:
Just to be more explicit, I'd say something like: "The structure passed to ansible as extra vars is:"